server/asset/eth/eth.go

```diff
@@ -189,7 +209,34 @@ func (eth *Backend) ValidateSecret(secret, contract []byte) bool {
 
 // Synced is true if the blockchain is ready for action.
 func (eth *Backend) Synced() (bool, error) {
-	return false, notImplementedErr
+	// node.SyncProgress will return nil both before syncing has begun and
+	// after it has finished. In order to discern when syncing has begun,
+	// wait for at least one change in block count, then defer to
+	// node.SyncProgress for the lifetime of the eth backend.
+	syncingStarted := atomic.LoadUint32(&eth.syncingStarted)
+	if syncingStarted != 1 {
+		ibn := atomic.LoadUint64(&eth.initBlockNum)
+		bn, err := eth.node.blockNumber(eth.ctx)
+		if err != nil {
+			return false, err
+		}
+		if ibn == 0 {
+			atomic.StoreUint64(&eth.initBlockNum, bn)
```

Comment:
Should we set the `initBlockNum` in `(*Backend).Connect`?

Reply:
Looks like took a different direction, and no longer watching the block number.